### DISCUSSION TITLE
Images displaying

### DISCUSSION THREAD

**Discussion by Almaz5200** [wojciech-kulik/xcodebuild.nvim#151] [Ideas]
Thought you can’t display images in neovim, but turns out it’s possible with this plugin https://github.com/3rd/image.nvim

could be cool to display snapshots like that, and possibly images from assets that way? I remember AppCode displayed them near the line number when you mention image’s name

**Comment by wojciech-kulik** [comment]
I was trying to do that, but it highly depends on your terminal, so it wouldn't be a universal solution. The plugin you attached works only with Kitty. On top of that, if you use a multiplexer like tmux or something else, it can break this functionality as well.

**Reply by Almaz5200** [reply]
Got it, thanks! Too bad, it would be so cool. Guess I'd need Emacs do to that :D

**Comment by wojciech-kulik** [comment]
Now we have snacks.nvim, which integrates well with images. I use this keymap to show snapshots with a preview:

```lua
    vim.keymap.set("n", "<leader>xs", function()
      local path = vim.fn.getcwd() .. "/.nvim/xcodebuild/failing-snapshots"
      if vim.fn.isdirectory(path) ~= 0 then
        Snacks.picker.files({
          dirs = { path },
          confirm = function(data)
            vim.fn.jobstart({"qlmanage", "-p", data.finder.items[1].file}, {
              detach = true,
              on_exit = function() end,
            })

            vim.defer_fn(function()
              vim.fn.system("open -a qlmanage")
            end, 200)
          end,
        })
      else
        Snacks.notify.info("No Failing Snapshots")
      end
    end, { desc = "Show Failing Snapshots" })
```
